src/components/CohortBuilder/ParticipantsTableView/SaveSetModal.tsx

```diff
@@ -155,24 +151,18 @@ const SaveSetModal: FunctionComponent<Props> = (props) => {
           label="Name"
           name="nameSet"
           hasFeedback
-          validateStatus={displayHelp || hasError ? 'error' : 'success'}
-          help={
-            displayHelp ? displayHelpMessage : 'Letters, numbers, hyphens (-), and underscores (_)'
-          }
+          validateStatus={hasError ? 'error' : 'success'}
+          help={error ? error.message : 'Letters, numbers, hyphens (-), and underscores (_)'}
           rules={[
             () => ({
               validator: (_, value) => {
                 if (hasError) {
                   reInitializeState();
                 }
-                setDisplayHelp(false);
-                setDisplayHelpMessage('');
                 const { msg, err } = validateNameSetInput(value);
-                isSaveButtonDisabled();
                 setHasError(err);
```

Comment:
why not set the error only when there is an error? Couldn't we call 'setHasError' in the if (err) block?